lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -1477,6 +1436,16 @@ protected getCachedTokenInternal(scopes : Array<string> , user: User): CacheResu
     this.loadIframeTimeout(urlNavigate, "msalIdTokenFrame", this.clientId);
   }
 
+  private updateAcquireTokenCache(authenticationRequest: AuthenticationRequestParameters, user: User) {
+    const userIdentifier = user && user.userIdentifier || Constants.no_user;
+    const acquireTokenUserKey = Constants.acquireTokenUser + Constants.resourceDelimeter + userIdentifier + Constants.resourceDelimeter + authenticationRequest.state;
+    const authorityKey = Constants.authority + Constants.resourceDelimeter + authenticationRequest.state;
+
+    this._cacheStorage.setItem(acquireTokenUserKey, JSON.stringify(user));
+    this._cacheStorage.setItem(authorityKey, authenticationRequest.authority, this.storeAuthStateInCookie);
+    this._cacheStorage.setItem(Constants.nonceIdToken, authenticationRequest.nonce);
```

Comment:
The storeAuthStateInCookie flag should be passed in here as well, correct?

Reply:
`storeAuthStateInCookie` was only provided for the authority key in all the previous code sections that this function is replacing.  This change is only about consolidating code / eliminating duplicate code while keeping the same functionality.  Changes to the functionality, such as including `storeAuthStateInCookie` for other storage values, is a discussion outside the scope of this change.

Reply:
We are using the storeAuthStateInCookie for the nonceIdToken parameter in line 1076 of the file before your changes. We are most likely going to refactor the usage for this functionality in the near future, but for now we will most likely need to pass this here as well.

Reply:
@manoj-rath @pkanher617 Added a commit that uses `storeAuthStateInCookie` for the nonce.